packages/core/src/builder.class.ts

```diff
@@ -984,6 +994,20 @@ export class Builder {
 
   static registerAction(action: Action) {
     this.actions.push(action);
+
+    if (Builder.isBrowser) {
+      const actionClone = JSON.parse(JSON.stringify(action));
+      if(action.action) {
+        actionClone.action = action.action.toString();
```

Comment:
In the `Action` type, the `action` property is a function type, but here we are setting it as a string. Is this intended ?

Reply:
Yes... function cannot be passed in a postMessage()

Reply:
Can you try these instead?
```suggestion
    const actionClone = {
      ...action,
      action: action.action?.toString(),
    };
```